libretro-common/include/queues/generic_queue.h

```diff
@@ -0,0 +1,208 @@
+/* Copyright  (C) 2010-2020 The RetroArch team
+ *
+ * ---------------------------------------------------------------------------------------
+ * The following license statement only applies to this file (generic_queue.h).
+ * ---------------------------------------------------------------------------------------
+ *
+ * Permission is hereby granted, free of charge,
+ * to any person obtaining a copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef __LIBRETRO_SDK_GENERIC_QUEUE_H
+#define __LIBRETRO_SDK_GENERIC_QUEUE_H
+
+#include <retro_common_api.h>
+
+#include <boolean.h>
+#include <stddef.h>
+
+RETRO_BEGIN_DECLS
+
+/**
+ * Represents a generic queue. Can contain any number of elements. Each element contains
+ * a value of type "void *". Can be used as a FIFO or LIFO queue.
+ */
+typedef struct generic_queue generic_queue_t;
+
+/**
+ * Represents an iterator for iterating over a queue.
+ */
+typedef struct generic_queue_iterator generic_queue_iterator_t;
+
+/**
+ * Creates a new queue with no elements.
+ *
+ * @return New queue
+ */
+generic_queue_t *generic_queue_new();
```

Comment:
Same thing here, and probably in some other places as well.

Reply:
Updated.